include/deal.II/numerics/vector_tools_project.templates.h

```diff
@@ -437,11 +434,9 @@ namespace VectorTools
                                     q_boundary,
                                     project_to_boundary_first);
 
-      const IndexSet &          locally_owned_dofs = dof.locally_owned_dofs();
-      IndexSet::ElementIterator it                 = locally_owned_dofs.begin();
-      for (; it != locally_owned_dofs.end(); ++it)
-        ::dealii::internal::ElementAccess<VectorType>::set(work_result(*it),
-                                                           *it,
+      for (const auto &it : dof.locally_owned_dofs())
+        ::dealii::internal::ElementAccess<VectorType>::set(work_result(it),
+                                                           it,
```

Comment:
```suggestion
      for (const auto dof : dof.locally_owned_dofs())
        ::dealii::internal::ElementAccess<VectorType>::set(work_result(dof),
                                                           dof,
```
For range-based for loops written like this, you are actually accessing the elements of the `IndexSet` container, and no longer explicitly through iterators. That's why you don't need to de-reference anymore.

Since the elements in an `IndexSet` are of type `size_type`, you also don't need to work with references, but can use the values themselves.